src/three-idx.ts

```diff
@@ -38,4 +82,92 @@ export class ThreeIDX {
+    this.docs[authLink] = await this.ceramic.createDocument('account-link', {
+      metadata: { owners: [authLink] },
+    })
+    await this.docs[authLink].change({ content: authEntry.linkProof })
+    await this.ceramic.pin.add(this.docs[authLink].id)
+    const tmpEntry = Object.assign({}, authEntry)
+    delete tmpEntry.linkProof
+    return { [this.docs[authLink].id]: tmpEntry }
+  }
+
+  /**
+   * Create a new IDX structure that has a given authEntry in it's keychain.
+   */
+  async createIDX(authEntry: NewAuthEntry): Promise<void> {
+    const entry = await this.createAuthMapEntry(authEntry)
+    this.docs[CDefs.authKeychain] = await this.ceramic.createDocument('tile', {
+      metadata: { owners: [this.DID] },
+      content: entry,
+    })
+    this.docs.idx = await this.ceramic.createDocument('tile', {
+      metadata: { owners: [this.DID] },
+      content: { [CDefs.authKeychain]: this.docs[CDefs.authKeychain].id },
+    })
+    await this.docs.threeId.change({
+      content: Object.assign(this.docs.threeId.content, { idx: this.docs.idx.id }),
+    })
+    await this.pinAllDocs()
+  }
+
+  /**
+   * Returns the encrypted JWE for the given authLink
+   */
+  async loadIDX(authLink: string): Promise<EncData | null> {
+    this.docs[authLink] = await this.ceramic.createDocument('account-link', {
+      metadata: { owners: [authLink] },
+    })
+    const did = this.docs[authLink].content
+    if (!did) return null
+    this.docs.threeId = await this.ceramic.loadDocument(`ceramic://${did.split(':')[2] as string}`)
+    const idxDocid = this.docs.threeId.content.idx
+    if (!idxDocid) return null
+    this.docs.idx = await this.ceramic.loadDocument(idxDocid)
+    const authKeychainDocid = this.docs.idx.content[CDefs.authKeychain]
+    if (!authKeychainDocid) return null
+    this.docs[CDefs.authKeychain] = await this.ceramic.loadDocument(authKeychainDocid)
+    const linkDocid = this.docs[authLink].id
+    return this.docs[CDefs.authKeychain].content[linkDocid].data
+  }
+
+  /**
+   * Adds a new AuthEntries to the Auth keychain.
+   */
+  async addAuthEntries(authEntries: Array<NewAuthEntry>): Promise<void> {
+    const mapEntries = await Promise.all(authEntries.map(this.createAuthMapEntry.bind(this)))
+    const content = Object.assign({}, this.docs[CDefs.authKeychain].content)
+    const newContent = mapEntries.reduce((acc, entry) => Object.assign(acc, entry), content)
+    await this.docs[CDefs.authKeychain].change({ content: newContent })
+  }
+
+  /**
+   * Returns all public keys that is in the auth keychain.
+   */
+  getAllAuthEntries(): Array<AuthEntry> {
+    if (!this.docs[CDefs.authKeychain]) return []
+    const content = Object.assign({}, this.docs[CDefs.authKeychain].content)
+    return Object.keys(content).map((authLink: string): AuthEntry => content[authLink] as AuthEntry)
+  }
+
+  async pinAllDocs(): Promise<void> {
+    await Promise.all(
+      Object.entries(this.docs).map(async ([_, doc]) => {
```

Comment:
Seems this could be replaced by `Object.values(this.docs).map(async (doc) => {` as the key is not used